ca_qc_sainte_anne_de_bellevue/people.py

```diff
@@ -8,24 +6,51 @@
 
 class SainteAnneDeBellevuePersonScraper(CanadianScraper):
     def scrape(self):
+        def decode_email(e):
+            de = ""
+            k = int(e[:2], 16)
+
+            for i in range(2, len(e) - 1, 2):
+                de += chr(int(e[i : i + 2], 16) ^ k)
+
+            return de
+
         page = self.lxmlize(COUNCIL_PAGE)
 
-        councillors = page.xpath('//div[@class="block text"]')
+        councillors = page.xpath('//div[@class="col-md-12"]')[0]
         assert len(councillors), "No councillors found"
-        for councillor in councillors:
-            name = councillor.xpath('.//div[@class="content-writable"]//strong/text()')[0]
-            district = councillor.xpath(".//h2/text()")[0]
 
-            if "Maire" in district:
-                district = "Sainte-Anne-de-Bellevue"
-                role = "Maire"
-            else:
-                district = "District {}".format(re.search(r"\d+", district)[0])
-                role = "Conseiller"
+        roles_and_districts = councillors.xpath(".//h2/text()")
+        roles = []
```

Comment:
This is a complicated and unusual way of doing this. Try to do a single loop that iterates on `//p[a[contains(@href, "@")]]`, and then does `.//preceding-sibling::h2` to get the matching heading. That way, you're not populating 4 arrays.